tests/test_daemon.cpp

```diff
@@ -526,6 +573,83 @@ TEST_P(DaemonCreateLaunchTestSuite, adds_pollinate_user_agent_to_cloud_init_conf
     send_command({GetParam()});
 }
 
+TEST_P(DaemonCreateLaunchTestSuite, workflow_found_passes_expected_data)
+{
+    auto mock_factory = use_a_mock_vm_factory();
+    auto mock_image_vault = std::make_unique<NiceMock<mpt::MockVMImageVault>>();
+    auto mock_workflow_provider = std::make_unique<NiceMock<mpt::MockVMWorkflowProvider>>();
+
+    static constexpr int num_cores = 4;
+    const mp::MemorySize mem_size{"4G"};
+    const mp::MemorySize disk_space{"25G"};
+    const std::string release{"focal"};
+    const std::string remote{"release"};
+
+    EXPECT_CALL(*mock_factory, create_virtual_machine(_, _))
+        .WillOnce([&mem_size, &disk_space](const mp::VirtualMachineDescription& vm_desc, auto&) {
+            EXPECT_EQ(vm_desc.num_cores, num_cores);
+            EXPECT_EQ(vm_desc.mem_size, mem_size);
+            EXPECT_EQ(vm_desc.disk_space, disk_space);
+
+            return std::make_unique<mpt::StubVirtualMachine>();
+        });
+
+    EXPECT_CALL(*mock_image_vault, fetch_image(_, _, _, _))
+        .WillOnce(Invoke([&release, &remote](const mp::FetchType& fetch_type, const mp::Query& query,
```

Comment:
Do you need the `Invoke()` here (and further down) for some reason? I thought just putting a lambda in `WillOnce()` would work?

ref.: https://google.github.io/googletest/gmock_cheat_sheet.html#using-a-function-functor-or-lambda-as-an-action

Reply:
You're right, I think I got on a copy & pasta kick and just threw the `Invoke` in these places without prejudice :wink: Will remove these.

Reply:
Ok, these have been removed for this PR.